app/services/course/assessment/submission/monitoring_service.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+class Course::Assessment::Submission::MonitoringService
+  class << self
+    def for(submission, assessment)
+      new(submission, assessment) if assessment.monitor_id?
+    end
+
+    def continue_listening_from(assessment, creator_ids)
+      sessions_from(assessment, creator_ids)&.update_all(status: :listening)
+    end
+
+    def destroy_all_by(assessment, creator_ids)
```

Comment:
Is there a risk where submission destroy is successful, but the session is not? Then we have an orphaned session.

Also, what will happen to bulk submissions deletion triggered by `Course::Assessment::Submission::DeletingJob`?

Reply:
Actually, we can just ignore this for now. Since heartbeats do not have `dependent: :destroy` on session destroy, sessions cannot be destroyed upon submission delete.

I'll just stop the session on submission delete for now. This means that each submission can only have exactly one session ever.